test/parallel/test-permission-deny-fs-read.js

```diff
@@ -0,0 +1,325 @@
+// Flags: --experimental-permission --allow-fs-read=* --allow-fs-write=*
+'use strict';
+
+const common = require('../common');
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+
+const assert = require('assert');
+const fs = require('fs');
+const fixtures = require('../common/fixtures');
+const tmpdir = require('../common/tmpdir');
+const path = require('path');
+
+const blockedFile = fixtures.path('permission', 'deny', 'protected-file.md');
+const relativeProtectedFile = './test/fixtures/permission/deny/protected-file.md';
+const absoluteProtectedFile = path.resolve(relativeProtectedFile);
+const blockedFolder = tmpdir.path + '/';
+const regularFile = __filename;
+
+{
+  tmpdir.refresh();
+  assert.ok(process.permission.deny('fs.read', [blockedFile, blockedFolder]));
+}
+
+// fs.readFile
+{
+  assert.throws(() => {
+    fs.readFile(blockedFile, () => {});
+  }, common.expectsError({
+    code: 'ERR_ACCESS_DENIED',
+    permission: 'FileSystemRead',
+    resource: blockedFile,
+  }));
+  assert.throws(() => {
+    fs.readFile(relativeProtectedFile, () => {});
+  }, common.expectsError({
+    code: 'ERR_ACCESS_DENIED',
+    permission: 'FileSystemRead',
+    resource: absoluteProtectedFile,
+  }));
+  assert.throws(() => {
+    fs.readFile(blockedFolder + 'anyfile', () => {});
+  }, common.expectsError({
+    code: 'ERR_ACCESS_DENIED',
+    permission: 'FileSystemRead',
+    resource: blockedFolder + 'anyfile',
+  }));
+
+  // doesNotThrow
+  fs.readFile(regularFile, () => {});
+}
+
+// fs.createReadStream
+{
+  assert.rejects(() => {
+    return new Promise((_resolve, reject) => {
+      const stream = fs.createReadStream(blockedFile);
+      stream.on('error', reject);
+    });
+  }, common.expectsError({
+    code: 'ERR_ACCESS_DENIED',
+    permission: 'FileSystemRead',
+    resource: blockedFile,
+  }));
```

Comment:
Ditto on all the other `assert.rejects` calls.
```suggestion
  })).then(common.mustCall());
```